### PR TITLE
Create multi-stage Dockerfile to minimize final Docker image size

### PR DESCRIPTION
**Pull request by lordgamez**
Using the latest released Docker image open62541/open62541 in test environments could be cumbersome due to its size which is over 500MBs. Creating a multi-stage Dockerfile can solve this issue by moving only all the built binaries and runtime dependencies to the final image build stage. This way the final image size can be lowered to 30MBs.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,36 +1,56 @@
...
+COPY --from=build /usr/local/lib64/libopen62541* /usr/local/lib64/
+COPY --from=build /usr/local/lib64/cmake/open62541/ /usr/local/lib64/cmake/open62541/
+COPY --from=build /usr/local/lib64/pkgconfig/open62541.pc /usr/local/lib64/pkgconfig/open62541.pc
+COPY --from=build /usr/local/share/open62541/tools/certs /usr/local/share/open62541/tools/certs
+COPY --from=build /usr/local/include/open62541/ /usr/local/include/open62541/
+COPY --from=build /usr/local/include/ms_stdint.h /usr/local/include/ms_stdint.h
+COPY --from=build /usr/local/include/ziptree.h /usr/local/include/ziptree.h
+COPY --from=build /usr/local/include/aa_tree.h /usr/local/include/aa_tree.h
+COPY --from=build /usr/local/bin/ua_server_ctt.exe /usr/local/bin/ua_server_ctt.exe
+COPY --from=build /usr/local/bin/ua_client /usr/local/bin/ua_client
```
Comment (JanSurft) [inline docker/Dockerfile:51]: We can get rid of all the explicit COPY layers and just copy over all necessary collected files in a single mount-and-copy layer.

```suggestion
RUN --mount=src=/release,dst=/release,from=collect \
    cp -r /release/* /
```

**Review by JanSurft — CHANGES_REQUESTED** [review]
Great work thanks a lot for showing us how to integrate multi-stage docker builds! I added some suggestions for change also to reduce the number of layers in the `server` stage.
Happy to see your comments about that.

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,36 +1,56 @@
-FROM alpine:3.10
+FROM alpine:3.13.6 AS build
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply docker/Dockerfile:2]: Added tagging in 946115aab12994a36174c0b0877ca7cc73cf0768

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,36 +1,56 @@
...
-RUN python3 /opt/open62541/tools/certs/create_self-signed.py /opt/open62541/pki/created
+RUN apk add --no-cache python3-dev linux-headers openssl && rm -rf /var/cache/apk/* && \
+    pip3 install --no-cache-dir netifaces==0.10.9 && \
+    mkdir -p /opt/open62541/pki/created && \
+    python3 /opt/open62541/tools/certs/create_self-signed.py /opt/open62541/pki/created
 
 
+FROM alpine:3.13.6 AS server
+
+RUN apk add --no-cache mbedtls-dev && \
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply docker/Dockerfile]: Good catch, changed in 946115aab12994a36174c0b0877ca7cc73cf0768

File: docker/Dockerfile (hunk 1 above)
Comment (lordgamez) [reply docker/Dockerfile]: Changed in 946115aab12994a36174c0b0877ca7cc73cf0768

**Comment by jpfr** [comment]
Great, thanks!

**Pull request merged**